### DISCUSSION TITLE
Hope to have support for a 16-core CPU

### DISCUSSION THREAD

**Discussion by Lkskywalker** [chidiwilliams/buzz#1020] [Ideas]
My CPU model is 7950X. When I run "Buzz," it only utilizes 60% of the CPU. Hope to optimize it to increase utilization.
![屏幕截图 2024-12-10 054413](https://github.com/user-attachments/assets/223473ed-99be-4a93-a6af-7abba0e1fe32)

**Comment by raivisdejus** [comment]
For Whisper.cpp you can use `BUZZ_WHISPERCPP_N_THREADS`, please see Advanced preferences https://chidiwilliams.github.io/buzz/docs/preferences#advanced-preferences

Full CPU utilization may not give faster results as if we transcribe parts of the audio in parallel the results of those parallel processes will need to be merged to get the final output. The defaults used are selected to give the fastest overall results. 

To get the fastest possible results Whisper will use real CPU cores, not threads, so that may be the reason you see partial CPU utilization.
